projects/packages/stats/src/class-wpcom-stats.php

```diff
@@ -231,8 +231,12 @@ public function get_views_by_country( $args = array() ) {
 	public function get_followers( $args = array() ) {
 
 		$this->resource = 'followers';
+		$followers      = $this->fetch_stats( $args );
 
-		return $this->fetch_stats( $args );
+		// run an action which allows us to fire a recommendation.
+		do_action( 'jetpack_followers_fetched', $followers );
```

Comment:
I wonder if there's another place that we can trigger an action to evaluate follower counts.
Where do we allow new subscribers, could we trigger an action each time someone subscribes to the blog?

This works OK, but it seems a bit out of place in this utility getter function.

Reply:
I had a good look through here: https://developer.jetpack.com/hooks/

> This works OK, but it seems a bit out of place in this utility getter function.

Maybe we need to have a webhook which WP.com fires to notify sites when they hit milestones.

Reply:
> This works OK, but it seems a bit out of place in this utility getter function.

Other options:-

- Keep it as it is, and it triggers when the subscriber counts are fetched.
- Do a timed cron based check where we call the WP.com API, would rely on the cronjob being triggered.
- Do something else, there's Achievements, which can email a notification & show a badge when milestones are hit:  fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Snpuvrirzragf.cuc%3Se%3Qn7p82158%23658-og

Reply:
@jboland88 - for when I pick this back up. What are your thoughts here?

Reply:
@mikestottuk - IMO the way that you have it now is fine. There may be a better solution out there, however, I think the other solutions are likely going to be more complex to implement.

If this recommendation does well and we want to improve how it is triggered, we could look at iterating in a later PR.